plugins/header_rewrite/conditions.cc

```diff
@@ -1373,3 +1374,50 @@ ConditionInbound::append_value(std::string &s, const Resources &res, NetworkSess
     s += zret;
   }
 }
+
+ConditionStringLiteral::ConditionStringLiteral(const std::string &v)
+{
+  TSDebug(PLUGIN_NAME_DBG, "Calling CTOR for ConditionStringLiteral");
+  _literal = v;
+}
+
+void
+ConditionStringLiteral::append_value(std::string &s, const Resources &res)
+{
+  s += _literal;
+  TSDebug(PLUGIN_NAME, "Appending '%s' to evaluation value -> %s", _literal.c_str(), s.c_str());
+}
+
+bool
+ConditionStringLiteral::eval(const Resources &res)
+{
+  std::string s;
+
+  append_value(s, res);
+  TSDebug(PLUGIN_NAME, "Evaluating StringLiteral");
+
+  return static_cast<const MatcherType *>(_matcher)->test(s);
```

Comment:
Why does it have to copy this string? Couldn't the matcher test() directly on _literal ?